File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTObjectCodedPropertyDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTObjectDef } from "./TTObjectDef";
import { TTDataType } from "../../DataDictionary/TTDataType";

export class TTObjectCodedPropertyDef {
    public ObjectDef: TTObjectDef;
    public ObjectDefID: Guid;
    public CodedPropertyDefID: Guid;
    public Name: string;
    public Description: string;
    public DataType: TTDataType;
    public DataTypeID: Guid;
    public GetScript: string;
    public SetScript: string;
    public ToString(): string {
        return null;
    }
}

File: AtlasModel/Enterprise/TTObjectCodedPropertyDef.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTObjectCodedPropertyDef
    {
        public Guid CodedPropertyDefId { get; set; }
        public Guid ObjectDefId { get; set; }
        public string Name { get; set; }
        public Guid DataTypeId { get; set; }
        public string GetScript { get; set; }
        public string SetScript { get; set; }
        public short CheckOutStatus { get; set; }
        public string Description { get; set; }
    }
}